Warehouse/WarehouseImplementation.cpp

```diff
@@ -714,12 +721,12 @@ namespace WPEFramework
             return Core::ERROR_NONE;
         }
 
-        void WarehouseImplementation::dispatchEvent(Event event, const JsonValue &params)
+        void WarehouseImplementation::dispatchEvent(Event event, const JsonObject &params)
         {
             Core::IWorkerPool::Instance().Submit(Job::Create(this, event, params));
         }
 
-        void WarehouseImplementation::Dispatch(Event event, const JsonValue params)
+        void WarehouseImplementation::Dispatch(Event event, const JsonObject params)
```

Comment:
Pls make the params as pass by reference. Anytime we are passing large objects by value can be expensive in embedded devices.